impure path when trying to build opencv with cuda

## Issue description
running nix-shell . with default.nix that sets enableCuda to true causes opencv to fail building with
 
impure path `/lib64/libz.so' used in link
collect2: error: ld returned 1 exit status


### Steps to reproduce
nix-shell .

##### Default.nix:
```
let
  pkgs = import <nixpkgs> {};
  stdenv = pkgs.stdenv;
  cuda-opencv3 = pkgs.opencv3.override{
enableCuda = true;
};
in rec {
  examplecuda = stdenv.mkDerivation rec {
    name = "example-cuda";
    src = ./.;
    buildInputs = [pkgs.cmake cuda-opencv3 pkgs.cudatoolkit pkgs.linuxPackages.nvidia_x11 pkgs.makeWrapper ];

    preBuild = ''
      export CUDA_PATH="${pkgs.cudatoolkit}"
    '';

    installPhase = ''
      mkdir -p "$out/bin"
      cp -r ./bin/*/*/release/* "$out/bin"

      for file in `ls $out/bin`; do
      	  echo "$file";
	  wrapProgram "$out/bin/$file" \
	              --prefix LD_LIBRARY_PATH ":" "${pkgs.linuxPackages.nvidia_x11}/lib"
      done
    '';


  };
}
```

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
Centos 7
3.10.0-327.22.2.el7.x86_64

* Nix version: (run `nix-env --version`)
nix-env (Nix) 1.11.4

* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
17.03pre97542.c1ecf36

--- a/pkgs/development/libraries/opencv/3.x.nix
+++ b/pkgs/development/libraries/opencv/3.x.nix
@@ -78,6 +78,12 @@ stdenv.mkDerivation rec {
         ln -s "${bootdescFiles}/$name" "$NIX_BUILD_TOP/opencv_contrib/xfeatures2d/src/$name"
       done
     '');
+
+  # This prevents cmake from using libraries in impure paths (which causes build failure on non NixOS)
+  postPatch = ''
+    sed -i '/Add these standard paths to the search paths for FIND_LIBRARY/,/^\s*$/{d}' CMakeLists.txt
+  '';
+
   preConfigure =
     (let version  = "20151201";
          md5      = "808b791a6eac9ed78d32a7666804320e";

--- a/pkgs/development/libraries/opencv/default.nix
+++ b/pkgs/development/libraries/opencv/default.nix
@@ -35,6 +35,11 @@ stdenv.mkDerivation rec {
       ./no-build-info.patch
     ];
 
+  # This prevents cmake from using libraries in impure paths (which causes build failure on non NixOS)
+  postPatch = ''
+    sed -i '/Add these standard paths to the search paths for FIND_LIBRARY/,/^\s*$/{d}' CMakeLists.txt
+  '';
+
   outputs = [ "out" "dev" ];
 
   buildInputs =

